File: TransCoder_tokenized_test_set_functions/python/ROPES_DATA_STRUCTURE_FAST_STRING_CONCATENATION.py
def concatenate ( a, b, c, n1, n2 ) :
    i = - 1
    for i in range ( n1 ) :
        c [ i ] = a [ i ]
    for j in range ( n2 ) :
        c [ i ] = b [ j ]
        i += 1
    

File: TransCoder_tokenized_test_set_functions/cpp/ROPES_DATA_STRUCTURE_FAST_STRING_CONCATENATION.cpp
void concatenate ( char a [ ], char b [ ], char c [ ], int n1, int n2 ) {
  int i;
  for ( i = 0;
  i < n1;
  i ++ ) c [ i ] = a [ i ];
  for ( int j = 0;
  j < n2;
  j ++ ) c [ i ++ ] = b [ j ];
  c [ i ] = '\0';
}